### PR TITLE
Add auth support

### LINKED ISSUE #18: Add TLS database connection support

### PR DESCRIPTION
**Pull request by EinKrebs**
Enables connection to remote databases with various settings.
This is properly done version of my previous pull request.

Resolves #18

### CODE REVIEW

**Review by pavolloffay — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -14,11 +14,11 @@ build:
 
 .PHONY: run
 run:
-	docker run --rm --name jaeger -e JAEGER_DISABLED=false --link some-clickhouse-server -it -u ${shell id -u} -p16686:16686 -p14250:14250 -p14268:14268 -p6831:6831/udp -v "${PWD}:/data" -e SPAN_STORAGE_TYPE=grpc-plugin jaegertracing/all-in-one:${JAEGER_VERSION} --query.ui-config=/data/jaeger-ui.json --grpc-storage-plugin.binary=/data/jaeger-clickhouse-$(GOOS)-$(GOARCH) --grpc-storage-plugin.configuration-file=/data/config.yaml --grpc-storage-plugin.log-level=debug
+	docker run --rm --name jaeger -e JAEGER_DISABLED=true -it -u ${shell id -u} --network="host" -p16686:16686 -p14250:14250 -p14268:14268 -p6831:6831/udp -v "${PWD}:/data" -e SPAN_STORAGE_TYPE=grpc-plugin jaegertracing/all-in-one:${JAEGER_VERSION} --query.ui-config=/data/jaeger-ui.json --grpc-storage-plugin.binary=/data/jaeger-clickhouse-$(GOOS)-$(GOARCH) --grpc-storage-plugin.configuration-file=/data/config.yaml --grpc-storage-plugin.log-level=debug
 
 .PHONY: run-hotrod
 run-hotrod:
-	docker run --rm  --link jaeger --env JAEGER_AGENT_HOST=jaeger  --env JAEGER_AGENT_PORT=6831  -p8080:8080 jaegertracing/example-hotrod:${JAEGER_VERSION} all
```
Comment (chhetripradeep) [inline Makefile:21]: May i ask why we decided to change from docker networking to host networking  ?

Comment (chhetripradeep) [reply Makefile:21]: Also why do we publish ports when using host networking ? 

```
WARNING: Published ports are discarded when using host network mode
```

Comment (chhetripradeep) [reply Makefile:21]: It breaks setup on MacOS for me, since in MacOS docker runs inside a linux VM.

Comment (EinKrebs) [reply Makefile:21]: i used it to make run-hotrod compatible with using external storages, for wich I used "host" networking in run. But, I guess, it's better to get rid of "host" in Makefile.

Comment (EinKrebs) [reply Makefile:21]: I made new pull request bringing old makefile schemes back.

**Pull request merged**